tidy3d/components/simulation.py

```diff
@@ -749,8 +750,10 @@ def _warn_grid_size_too_small(cls, val, values):  # pylint:disable=too-many-loca
         return val
 
     @pydantic.validator("sources", always=True)
-    def _source_homogeneous(cls, val, values):
-        """Error if a plane wave or gaussian beam source is not in a homogeneous region."""
+    def _source_homogeneous_isotropic(cls, val, values):
+        """Error if a plane wave or gaussian beam source is not in a homogeneous and isotropic
```

Comment:
TFSF should also error if the box crosses any anisotropic region?

Reply:
true, added